bpf/mock/Makefile

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (C) 2021 Authors of Cilium
+
+check_helper_headers:
+	# Check if helpers.h, helpers_skb.h and helpers_xdp.h are up to date.
+	# Generate temp headers from lib/helpers.h, lib/helpers_skb.h and lib/helpers_xdp.h.
+	# Compare them to the headers within the current directory.
+	# The temp headers are removed after finishing.
+	sed -f helpers.sed ../include/bpf/helpers.h > helpers_temp.h;
+	sed -f helpers_skb.sed ../include/bpf/helpers_skb.h > helpers_skb_temp.h;
+	sed -f helpers_xdp.sed ../include/bpf/helpers_xdp.h > helpers_xdp_temp.h;
+	if cmp helpers_temp.h helpers.h; \
+	then \
+		echo mock/helpers.h is up to date!; \
+	else \
+		echo mock/helpers.h is out of date!; \
+		rm helpers_temp.h helpers_skb_temp.h helpers_xdp_temp.h; \
+		exit 1; \
+	fi; \
+	if cmp helpers_skb_temp.h helpers_skb.h; \
+	then \
+		echo mock/helpers_skb.h is up to date!; \
+	else \
+		echo mock/helpers_skb.h is out of date!; \
+		rm helpers_temp.h helpers_skb_temp.h helpers_xdp_temp.h; \
+		exit 1; \
+	fi; \
+	if cmp helpers_xdp_temp.h helpers_xdp_temp.h; \
+	then \
+		echo mock/helpers_xdp.h is up to date!; \
+	else \
+		echo mock/helpers_xdp.h is out of date!; \
+		rm helpers_temp.h helpers_skb_temp.h helpers_xdp_temp.h; \
+		exit 1; \
+	fi; \
+	rm helpers_temp.h helpers_skb_temp.h helpers_xdp_temp.h;
+
+generate_helper_headers:
+	# Auto-generate helpers.h, helpers_skb.h, and helpers_xdp.h.
+	# Take contents from bpf/helpers.h, bpf/helpers_skb.h and bpf/helpers_xdp.h.
+	# Prune to be ready to mock and save to the current directory.
+	sed -f helpers.sed ../include/bpf/helpers.h > helpers.h; \
+	sed -f helpers_skb.sed ../include/bpf/helpers_skb.h > helpers_skb.h; \
+	sed -f helpers_xdp.sed ../include/bpf/helpers_xdp.h > helpers_xdp.h;
+
+mock_helpers:
```

Comment:
I hit this error when I pull these changes:

```
$ make -C bpf/mock mock_helpers                                                                                                  
open cmock.tar: no such file or directory
Unable to find image 'cmock:latest' locally
docker: Error response from daemon: pull access denied for cmock, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
Error: No such container:path: helpers:/mocks/
Error: No such container: helpers
Unable to find image 'cmock:latest' locally
docker: Error response from daemon: pull access denied for cmock, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
Error: No such container:path: helpers_skb:/mocks/
Error: No such container: helpers_skb
Unable to find image 'cmock:latest' locally
docker: Error response from daemon: pull access denied for cmock, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
Error: No such container:path: helpers_xdp:/mocks/
Error: No such container: helpers_xdp
mv: cannot stat 'mocks/*': No such file or directory
rmdir: failed to remove 'mocks': No such file or directory
make: *** [Makefile:49: mock_helpers] Error 1
```

I think that we can resolve this by following the Documentation/Makefile examples a bit more closely. For instance:
* Import the relevant top-level makefiles so that the makefile variables like `CONTAINER_ENGINE` can be reused
* The "builder-image" target in Documentation/ prepares the image in the way that "build_docker_image" target in this PR tries to do. This makefile can do very similar commands, maybe just tagging this as cilium/bpf-mock instead of the cilium/docs-builder.
* The `DOCKER_CTR` environment variable then runs this container, making the local directiory available within the container without having to manually `docker cp` each file in. The `env` from the docker version of this is not necessary here (READTHEDOCS / SKIP_LINT are only relevant to Documentation/). But you could basically copy the rest of this variable declaration into this makefile + DOCKER_RUN so this can be reused in other targets.
* Targets like `mock_helpers` should depend on the target to build the image, so that we know it is available before trying to run commands.
* Rather than individually running each mock command as a dedicated line in the Makefile, we can refactor this into a dedicated script which is made available inside the container, so that there is only one docker run command, and it invokes the script; the script can be super simple like `Documentation/check-cmdref.sh`, basically just a couple of lines which in this case would be the `cmock ruby ...` commands.

Reply:
Thanks! Revised exactly as you suggest.